core/templates/xregistry_openapi_template.json

```diff
@@ -1,1068 +1,2042 @@
-              "description": "A number representing the version number of the resource.",
-              "format": "int64"
-            }
-          },
-          {
-            "$ref": "#/components/parameters/resource-description"
-          },
-          {
-            "$ref": "#/components/parameters/resource-documentation"
-          },
-          {
-            "$ref": "#/components/parameters/resource-labels"
-          }
-        ],
-        "summary": "Create or update a resource",
-        "description": "Create a resource. ",
-        "operationId": "put{%-resourceNameSingular-%}",
-        "requestBody": {
-          "description": "The resource to be created/updated",
-          "content": {
-            "application/octet-stream": {
-              "schema": {
-                "type": "string",
-                "format": "binary"
-              }
+        },
+        "/{%-groupNamePlural-%}/{groupid}/{%-resourceNamePlural-%}/{resourceid}/versions": {
+            "parameters": [
+                {
+                    "$ref": "#/components/parameters/groupid"
+                },
+                {
+                    "$ref": "#/components/parameters/resourceid"
+                },
+                {
+                    "$ref": "#/components/parameters/specversion"
+                },
+                {
+                    "$ref": "#/components/parameters/inline"
+                },
+                {
+                    "$ref": "#/components/parameters/filter"
+                },
+                {
+                    "$ref": "#/components/parameters/doc"
+                }
+            ],
+            "get": {
+                "operationId": "get{%-resourceNameSingular-%}VersionsAll",
+                "description": "Gets an optionally filtered collection of versions for a resource. Supports pagination ('skip', 'top') and filtering. (See spec: Retrieving all Versions)",
+                "parameters": [
+                    {
+                        "in": "query",
+                        "name": "skip",
+                        "description": "The number of versions to skip.",
+                        "required": false,
+                        "schema": { "type": "integer" }
+                    },
+                    {
+                        "in": "query",
+                        "name": "top",
+                        "description": "The maximum number of versions to return.",
+                        "required": false,
+                        "schema": { "type": "integer" }
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The collection of versions.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "type": "object",
+                                    "additionalProperties": {
```

Comment:
Just a pointer for me - why is the $ref inside `additionalProperties`?

Reply:
![image](https://github.com/user-attachments/assets/6279fab2-14a0-4284-b11b-77eec28acb88)

Is this the desired result?

Reply:
That is false. Correcting.